helper/resource/testing.go

```diff
@@ -1232,6 +1232,14 @@ func TestCheckModuleResourceAttrPair(mpFirst []string, nameFirst string, keyFirs
 }
 
 func testCheckResourceAttrPair(isFirst *terraform.InstanceState, nameFirst string, keyFirst string, isSecond *terraform.InstanceState, nameSecond string, keySecond string) error {
+	if nameFirst == nameSecond && keyFirst == keySecond {
```

Comment:
While I like the enhancement, I'm inclined to think in V1 maybe this should just be a warning like "you may be comparing an attribute with itself". We can make it an explicit error in V2.... tough one kind of a grey area on if this is a "breaking change" or a bugfix. I could be convinced either way.

Reply:
But yeah, would the warning ever get noticed in review/during development vs explicit error, it certainly is valuable in avoiding accidental merging of true == true when a PR is showing all tests passing.